### PR TITLE
chore(smime): handle signature verification early

### PR DESCRIPTION
**Pull request by st3iny**
This improves code quality and performance, as the full message fetch to verify the signature is reused to fetch body and all attachments (if requested).

### CODE REVIEW

**Comment by st3iny** [comment]
The codecov check is lying :cry: 

grafik

File: tests/Integration/IMAP/ImapMessageFetcherIntegrationTest.php [hunk 1]
```diff
@@ -114,6 +119,9 @@ public function testFetchMessageWithEncryptedMessage(): void {
 
 		$this->assertEquals("Just some encrypted test images.\n\n", $message->getPlainBody());
 		$this->assertCount(3, $message->attachments);
+		$this->assertTrue($message->isEncrypted());
+		// TODO: $this->assertTrue($message->isSigned());
```
Comment (ChristophWurst) [inline tests/Integration/IMAP/ImapMessageFetcherIntegrationTest.php:123]: todo?

Comment (st3iny) [reply tests/Integration/IMAP/ImapMessageFetcherIntegrationTest.php:123]: https://github.com/nextcloud/mail/issues/8080

File: lib/IMAP/ImapMessageFetcher.php [hunk 2]
```diff
@@ -139,6 +139,8 @@ public function fetchMessage(?Horde_Imap_Client_Data_Fetch $fetch = null): IMAPM
 			$this->hasAnyAttachment = $this->hasAttachments($structure);
 
 			$isEncrypted = $this->smimeService->isEncrypted($fetch);
+			$isOpaqueSigned = $structure->getType() === 'application/pkcs7-mime'
```
Comment (kesselb) [inline lib/IMAP/ImapMessageFetcher.php:142]: Can we add a fallback for application/x-pkcs7-mime?

**Review by kesselb — APPROVED** [review]
Nice

File: lib/IMAP/ImapMessageFetcher.php (hunk 2 above)
Comment (st3iny) [reply lib/IMAP/ImapMessageFetcher.php]: Done

**Pull request merged**